pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -963,10 +963,10 @@ func PublishAppInfoToZedCloud(ctx *zedagentContext, uuid string,
 		for _, ifname := range ifNames {
 			networkInfo := new(info.ZInfoNetwork)
 			networkInfo.LocalName = *proto.String(ifname)
-			ip, allocated, macAddr, ipAddrMismatch := getAppIP(ctx, aiStatus,
+			ips, allocated, macAddr, ipAddrMismatch := getAppIP(ctx, aiStatus,
 				ifname)
-			networkInfo.IPAddrs = make([]string, 1)
-			networkInfo.IPAddrs[0] = *proto.String(ip)
+			networkInfo.IPAddrs = make([]string, len(ips))
+			networkInfo.IPAddrs = ips
 			networkInfo.MacAddr = *proto.String(macAddr)
 			networkInfo.Up = allocated
```

Comment:
Should we have separate flags for v6 & v6?

Reply:
Currently we set this from DHCPv4.
We could potentially set it for DHCPv6 using the same mechanism. 
But to be able to set if for SLAAC (stateless address autoconfiguration) we'd reaaly need to be able to set it for each IP address *and* add code to use the DAD probe to detect the app instance is using it. I wonder whether it would make sense to have a IPv6Up flag which is set when we see any DAD probe from the app instance. If you think that makes sense then we can rename this IPv4Up and add that logic.
Otherwise it would still make sense to rename it to IPv4Up to make the semantics of the flag clearer.

Reply:
Since this flag is part of proto, we can do that following.
Create two new flags IPv4Up, IPv6Up
Set IPv4Up to true if there is an IPv4 address assigned
Set IPv6Up to true if there is a DAD probe seen with non link local address.

Would that work for now?

Reply:
Uyou can also rename Up to Ipv4UP (keep the same protobuf tag); just means that when the controller picks up the new proto definitions to need to rename things there as well. The name of the field in the proto file is a local matter; the wire encoding just cares about the tags and the datatypes. For things like renaming a boolean that makes sense to me.

Reply:
done